src/IO/UseSSL.cpp

```diff
@@ -2,23 +2,61 @@
 
 #include "config.h"
 
+#include <Common/Exception.h>
+
 #if USE_SSL
-#    include <Poco/Net/SSLManager.h>
+#    include <openssl/provider.h>
+#    include <openssl/crypto.h>
 #endif
 
 namespace DB
 {
+
+namespace ErrorCodes
+{
+    extern const int OPENSSL_ERROR;
+}
+
+#if USE_SSL
+OSSL_PROVIDER * DB::UseSSL::default_provider = nullptr;
+OSSL_PROVIDER * DB::UseSSL::legacy_provider = nullptr;
+#endif
+
 UseSSL::UseSSL()
 {
 #if USE_SSL
-    Poco::Net::initializeSSL();
+    if (ref_count++ == 0)
+    {
+        OPENSSL_init_crypto(OPENSSL_INIT_LOAD_CONFIG, nullptr);
```

Comment:
A lot more init stuff was called in Poco's OSSL initialization: https://github.com/ClickHouse/ClickHouse/blob/master/base/poco/Crypto/src/OpenSSLInitializer.cpp#L62-L114

Do we need to retain any of this?